plugins/modules/server_role.ps1

```diff
@@ -0,0 +1,129 @@
+        state = @{type = 'str'; required = $false; default = 'present'; choices = @('present', 'absent') }
+    }
+}
+
+$module = [Ansible.Basic.AnsibleModule]::Create($args, $spec, @(Get-LowlyDbaSqlServerAuthSpec))
+$sqlInstance, $sqlCredential = Get-SqlCredential -Module $module
+$roleName = $module.Params.role_name
+$loginName = $module.Params.login_name
+$ownerName = $module.Params.owner_name
+$state = $module.Params.state
+$checkMode = $module.CheckMode
+$module.Result.changed = $false
+
+try {
+    # Get server role status
+    try {
+        $server = Connect-DbaInstance -SqlInstance $sqlInstance -SqlCredential $sqlCredential
+        $existingServerRole = $server | Get-DbaServerRole -ServerRole $roleName
+    }
+    catch {
+        $module.FailJson("Error checking server role status.", $_.Exception.Message)
+    }
+
+    if ($state -eq "absent") {
+        if ($null -ne $existingServerRole) {
+            try {
+                $removeServerRoleSplat = @{
+                    SqlInstance = $sqlInstance
+                    SqlCredential = $sqlCredential
+                    ServerRole = $roleName
+                    WhatIf = $checkMode
+                    EnableException = $true
+                    Confirm = $false
+                }
+                $output = Remove-DbaServerRole @removeServerRoleSplat
+                if ($output.Status -eq "Success") {
+                    $module.Result.changed = $true
+                }
+                elseif ($output.Status -ne "Success") {
+                    $module.FailJson("Server role [$roleName] was not removed. " + $output.Status)
+                }
+            }
+            catch {
+                $module.FailJson("Deleting server role [$roleName] failed.", $_)
+            }
+        }
+
+        $module.ExitJson()
+    }
+    elseif ($state -eq "present") {
+        # Create server role
+        if ($null -eq $existingServerRole) {
+            try {
+                $newServerRoleSplat = @{
+                    SqlInstance = $sqlInstance
+                    SqlCredential = $sqlCredential
+                    ServerRole = $roleName
+                    WhatIf = $checkMode
+                    EnableException = $true
+                    Confirm = $false
+                }
+
+                if ($null -ne $ownerName) {
+                    $newServerRoleSplat.Owner = $ownerName
+                }
+                $output = New-DbaServerRole @newServerRoleSplat
+                $module.Result.changed = $true
+            }
+            catch {
+                $module.FailJson("Creating server role [$roleName] failed.", $_)
+            }
+        }
+        elseif ($existingServerRole.Owner -ne $ownerName) {
+            $existingServerRole.Owner = $ownerName
+            $existingServerRole.Alter()
+            $output = $existingServerRole
+            $module.Result.changed = $true
+        }
+
+        if ($null -ne $loginName) {
```

Comment:
@milanjelicic should we remove this part from this module? It can be implemented as another step in the playbook using the generic nonquery module.